server/aesdsocket.c

```diff
@@ -221,11 +390,28 @@ int main(int argc, char *argv[]) {
     if (daemon) {
         daemonize();
     }
+    
+    pthread_t timestamp_thread;
+    if (pthread_create(&timestamp_thread, NULL, append_timestamp, NULL) != 0) {
+        syslog(LOG_ERR, "Timestamp thread creation error");
+        cleanup_resources();
+        exit(EXIT_FAILURE);
+    }
 
-    listen_for_connections(custom_socket_fd);
-    accept_clients(custom_socket_fd);
+    while (!sig_exit) {
+        listen_for_connections(custom_socket_fd);
+        accept_clients(custom_socket_fd);
+    }
 
+    // Allow the timestamp thread to write the final timestamp
+    sleep(30);  // Adjust the sleep duration as needed
+
+    // Join completed threads after the main loop
+    join_completed_threads();
```

Comment:
@Suhas-Reddy-S what does it mean to only join completed threads before exiting and not as a part of the accept/listen loop?